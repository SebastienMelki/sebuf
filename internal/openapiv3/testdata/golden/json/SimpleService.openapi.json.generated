{"components":{"schemas":{"FieldViolation":{"description":"FieldViolation describes a single validation error for a specific field.","properties":{"description":{"description":"Human-readable description of the validation violation (e.g., 'must be a valid email address', 'required field missing')","type":"string"},"field":{"description":"The field path that failed validation (e.g., 'user.email' for nested fields). For header validation, this will be the header name (e.g., 'X-API-Key')","type":"string"}},"required":["field","description"],"type":"object"},"SimpleRequest":{"description":"Simple message for testing basic OpenAPI generation","properties":{"active":{"description":"Whether user is active","type":"boolean"},"age":{"description":"User's age","format":"int32","type":"integer"},"name":{"description":"User's name","type":"string"}},"type":"object"},"SimpleResponse":{"description":"Simple response message","properties":{"message":{"description":"Response message","type":"string"},"success":{"description":"Success indicator","type":"boolean"},"timestamp":{"description":"Response timestamp","format":"int64","type":"integer"}},"type":"object"},"ValidationError":{"description":"ValidationError is returned when request validation fails. It contains a list of field violations describing what went wrong.","properties":{"violations":{"description":"List of validation violations","items":{"$ref":"#/components/schemas/FieldViolation"},"type":"array"}},"required":["violations"],"type":"object"}}},"info":{"title":"SimpleService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/SimpleService/Create":{"post":{"description":"Create a simple resource","operationId":"Create","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/SimpleRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/SimpleResponse"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"Create","tags":["SimpleService"]}},"/SimpleService/Get":{"post":{"description":"Get a simple resource","operationId":"Get","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/SimpleRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/SimpleResponse"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"Get","tags":["SimpleService"]}}}}