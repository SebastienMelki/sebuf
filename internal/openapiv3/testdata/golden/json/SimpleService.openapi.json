{"components":{"schemas":{"SimpleRequest":{"description":"Simple message for testing basic OpenAPI generation","properties":{"active":{"description":"Whether user is active","type":"boolean"},"age":{"description":"User's age","format":"int32","type":"integer"},"name":{"description":"User's name","type":"string"}},"type":"object"},"SimpleResponse":{"description":"Simple response message","properties":{"message":{"description":"Response message","type":"string"},"success":{"description":"Success indicator","type":"boolean"},"timestamp":{"description":"Response timestamp","format":"int64","type":"integer"}},"type":"object"}}},"info":{"title":"SimpleService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/SimpleService/Create":{"post":{"description":"Create a simple resource","operationId":"Create","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/SimpleRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/SimpleResponse"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"Create","tags":["SimpleService"]}},"/SimpleService/Get":{"post":{"description":"Get a simple resource","operationId":"Get","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/SimpleRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/SimpleResponse"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"Get","tags":["SimpleService"]}}}}